Index: aaa_base/etc/init.d/random
================================================================================
--- aaa_base/etc/init.d/boot.clock
+++ aaa_base/etc/init.d/boot.clock
@@ -125,14 +125,6 @@
 		fi
 		;;
 	    esac
-	    #
-	    # Let zic set timezone - if present.
-	    #
-	    if test -n "$TIMEZONE" -a -x /usr/sbin/zic ; then
-		echo -n Setting up timezone data
-		/usr/sbin/zic -l $TIMEZONE
-		rc_status
-	    fi
 
 	    rc_status -v
 	    rc_exit
@@ -150,6 +142,10 @@
 	# Read out to hardware clock and for UTC calculate adjtime
 	# write back the system time later at reboot/shutdown time.
 	#
+	if rc_readonlyroot ; then
+	    # File /etc/adjtime is not writable yet
+	    USE_ADJFILE=no
+	fi
 	if test "$USE_ADJFILE" = yes ; then
 	    #
 	    # For UTC calculate adjtime
--- aaa_base/etc/init.d/random
+++ aaa_base/etc/init.d/random
@@ -35,8 +35,17 @@
         # Carry a random seed from start-up to start-up
         # Load and then save 512 bytes, which is the size of the entropy pool
         if test -f $random_seed ; then
-                cat $random_seed > /dev/urandom
-		rc_status
+		if tmp=$(/bin/mktemp -t ${random_seed##*/}.XXXXXXXXXX) && \
+		   cat $random_seed > $tmp && \
+		   cat $tmp 2>/dev/null > $random_seed; then
+		    cat $random_seed > /dev/urandom
+		    rc_status
+		else
+		    rc_status -v
+		    echo "Cannot write to $random_seed; refusing to seed" \
+			 "random number generator"
+		   rc_exit
+		fi
         else
                 > $random_seed
 		rc_status
